lib/topological_inventory/ansible_tower/operations/source.rb

```diff
@@ -10,58 +10,113 @@ class Source
         include Logging
         STATUS_AVAILABLE, STATUS_UNAVAILABLE = %w[available unavailable].freeze
 
-        attr_accessor :params, :context, :source_id
+        ERROR_MESSAGES = {
+          :authentication_not_found => "Authentication not found in Sources API",
+          :endpoint_not_found       => "Endpoint not found in Sources API",
+        }.freeze
+
+        attr_accessor :params, :request_context, :source_id
 
         def initialize(params = {}, request_context = nil)
-          @params  = params
-          @context = request_context
-          @source_id = nil
+          self.params          = params
+          self.request_context = request_context
+          self.source_id       = params['source_id']
         end
 
         def availability_check
-          source_id = params["source_id"]
-          unless source_id
-            logger.error("Missing source_id for the availability_check request")
-            return
-          end
+          return if params_missing?
 
-          source = ::SourcesApiClient::Source.new
-          source.availability_status = connection_check(source_id)
+          status, error_message = connection_status
 
-          begin
-            api_client.update_source(source_id, source)
-          rescue ::SourcesApiClient::ApiError => e
-            logger.error("Failed to update Source id:#{source_id} - #{e.message}")
-          end
-          logger.info("Source#availability_check completed: Source #{source_id} is #{source.availability_status}")
+          update_source_and_endpoint(status, error_message)
+
+          logger.info("Source#availability_check completed: Source #{source_id} is #{status}")
         end
 
         private
 
-        def connection_check(source_id)
-          endpoints = api_client.list_source_endpoints(source_id)&.data || []
-          endpoint = endpoints.find(&:default)
-          return STATUS_UNAVAILABLE unless endpoint
+        def required_params
+          %w[source_id]
+        end
 
-          endpoint_authentications = api_client.list_endpoint_authentications(endpoint.id.to_s).data || []
-          return STATUS_UNAVAILABLE if endpoint_authentications.empty?
+        def params_missing?
+          is_missing = false
+          required_params.each do |attr|
+            if (is_missing = params[attr].blank?)
+              logger.error("Source#availability_check - Missing #{attr} for the availability_check request")
+              break
+            end
+          end
 
-          auth_id = endpoint_authentications.first.id
-          auth = Core::AuthenticationRetriever.new(auth_id, identity).process
-          return STATUS_UNAVAILABLE unless auth
+          is_missing
+        end
+
+        def connection_status
+          return [STATUS_UNAVAILABLE, ERROR_MESSAGES[:endpoint]] unless endpoint
```

Comment:
I think this should be `ERROR_MESSAGES[:endpoint_not_found]` instead of `ERROR_MESSAGES[:endpoint]`